### PR TITLE
DAOS-3296 iosrv allow daos_io_server to start with oversubscribed targets

### PR DESCRIPTION
**Pull request by GitHuaKuang**
This is a new approach of DAOS-2799 to solve the CI master test timeout
    issue after it landed (reverted)

    Added a new env variable ENABLE_TARGET_OVERSUBSCRIBE
    to enable/disable target oversubscribing.

    By default, ENABLE_TARGET_OVERSUBSCRIBE is not set, or set to 0, and the current
    existing behave will be unchanged.

Change-Id: I7e6840d344bd81ddfe3c30666e99c28ea7bb6f00
Signed-off-by: Hua Kuang <hua.kuang@intel.com>

### CODE REVIEW

**Comment by GitHuaKuang** [comment]
https://jira.hpdd.intel.com/browse/DAOS-3903 is created to address the pi_bits mismatch issue exposed in CI test. Other than that, all test checks passed. Please give it one more review.

File: src/tests/ftest/io/nvme_io_verification.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ server_config:
  servers:
    bdev_class: nvme
    bdev_list: ["0000:5e:00.0","0000:5f:00.0"]
-   targets: 1
+   targets: 8
```
Comment (phender) [inline src/tests/ftest/io/nvme_io_verification.yaml]: The original intent of using `targets: 1` was to override the default (8).  Now that we will not need it, shouldn't we completely remove the `targets: <int>` entry, so the default is used?

This comment would apply to the other three yaml files in this PR

File: Jenkinsfile [hunk 2]
```diff
@@ -1259,6 +1259,8 @@ pipeline {
                                                test_tag=pr,-hw
                                            fi
                                            tnodes=$(echo $NODELIST | cut -d ',' -f 1-9)
+					   # set ENABLE_TARGET_OVERSUBSCRIBE env here
+					   export ENABLE_TARGET_OVERSUBSCRIBE=0
```
Comment (phender) [inline Jenkinsfile]: I'm unsure if it matters (other than to make code look cleaner), but is the indent off here?

File: Jenkinsfile [hunk 3]
```diff
@@ -1337,6 +1339,8 @@ pipeline {
                                                test_tag=pr,hw
                                            fi
                                            tnodes=$(echo $NODELIST | cut -d ',' -f 1-9)
+					   # set ENABLE_TARGET_OVERSUBSCRIBE env here
+					   export ENABLE_TARGET_OVERSUBSCRIBE=1
```
Comment (phender) [inline Jenkinsfile]: Same comment as above

File: src/tests/ftest/io/nvme_io_verification.yaml (hunk 1 above)
Comment (GitHuaKuang) [reply src/tests/ftest/io/nvme_io_verification.yaml]: done

File: Jenkinsfile (hunk 2 above)
Comment (GitHuaKuang) [reply Jenkinsfile]: For the sake of safety, in case the previous test set this env to 1, next test need to set it to 0.

Comment (GitHuaKuang) [reply Jenkinsfile]: the ident is accidently added due to the vi editor default setting. fixed.

File: Jenkinsfile (hunk 3 above)
Comment (GitHuaKuang) [reply Jenkinsfile]: fixed.

File: src/iosrv/init.c [hunk 4]
```diff
@@ -238,6 +241,17 @@ dss_tgt_nr_get(int ncores, int nr)
 	return nr_default;
 }
 
+static bool
+enable_target_oversubscribe(void)
+{
+	bool oversub = false;
+
+	if (getenv("ENABLE_TARGET_OVERSUBSCRIBE") == NULL)
+		return oversub;
```
Comment (liuxuezhao) [inline src/iosrv/init.c]: the above two lines is not needed as d_get_env_bool() can handle it.
and, tgt_oversub_enabled() maybe more align with naming rule.

Comment (GitHuaKuang) [reply src/iosrv/init.c]: good point - the whole function can be removed, use d_get_env_bool() directly.

File: Jenkinsfile [hunk 5]
```diff
@@ -1054,6 +1054,8 @@ pipeline {
                                                test_tag=pr,-hw
                                            fi
                                            tnodes=$(echo $NODELIST | cut -d ',' -f 1-9)
+                                           # set DAOS_TARGET_OVERSUBSCRIBE env here
+                                           export DAOS_TARGET_OVERSUBSCRIBE=0
```
Comment (brianjmurrell) [inline Jenkinsfile]: Why does this only apply to CI?

**Comment by GitHuaKuang** [comment]
It applies to other test too. For manual test, this env can be set manually enable/disable.
For CI, since multiple tests are running consecutively, the env set by previous test may need to
Be changed for current test, so, do it in Jenkinsfile to make sure it’s set properly.

Hua

**Pull request merged**